db/collection.go

```diff
@@ -341,13 +431,32 @@ func (t *Txn) Delete(ids ...core.InstanceID) error {
 
 // Has returns true if all IDs exists in the collection, false otherwise.
 func (t *Txn) Has(ids ...core.InstanceID) (bool, error) {
+	if err := t.collection.db.connector.Validate(t.token, true); err != nil {
```

Comment:
Here's an example of calling down into the `net` layer to check a `thread.Token`. The bool here means the access is read-only.